ui/src/stores/indexedDB.ts

```diff
@@ -70,13 +70,41 @@ export async function openDatabase(): Promise<IDBDatabase> {
     request.onsuccess = async (event) => {
       const db = (event.target as IDBOpenDBRequest).result;
       await ensureInitialState(db);
+      const general = await loadStateFromIndexedDB("general", db);
+      const shouldClearClipboard = await checkClipboardTimeout(general);
+
+      if (shouldClearClipboard && general?.clipboard) {
+        general.clipboard = {
+          nodes: [],
+          edges: [],
+          workflows: [],
+          copiedAt: null,
+        };
+        saveStateToIndexedDB({ clipboard: general.clipboard }, "general");
+      }
+
       resolve(db);
     };
 
     request.onerror = () => reject(request.error);
   });
 }
 
+async function checkClipboardTimeout(
+  state: GeneralState | null,
+): Promise<boolean> {
+  if (!state) {
+    return false;
+  }
+  if (
+    state?.clipboard?.copiedAt &&
+    Date.now() - state.clipboard.copiedAt > 1000 * 60 * 5
+  ) {
```

Comment:
If you can add the timeout period into constants.ts it'll be easier to change later if we want to.